tools/carveme/carveme.xml

```diff
@@ -0,0 +1,116 @@
+<tool id="carveme" name="CarveMe" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="21.05">
+    <description>Tool to reconstruct genome-scale metabolic models</description>
+    <macros>
+        <token name="@TOOL_VERSION@">1.6.5</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+        <xml name="biotools">
+            <xrefs>
+                <xref type="bio.tools">CarveMe</xref>
+            </xrefs>
+        </xml>
+    </macros>
+    <requirements>
+        <requirement type='package' version="@TOOL_VERSION@">carveme</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+    carve '$genome'
+    $input_format
+    $output_format
+    $universe
+    #if $diamond_args
+        --diamond-args '$diamond_args'
+    #end if
+    #if $universe_file
+        --universe-file '$universe_file'
+    #end if
+    #if $mediadb
+        --mediadb '$mediadb'
+    #end if
+    #if $soft
+        --soft '$soft'
+    #end if
+    #if $hard
+        --hard '$hard'
+    #end if
+    #if $reference
+        --reference '$reference'
+    #end if
+    --solver scip
+    --output $model
+    ]]></command>
+    <inputs>
+    <param name="genome" label="Genome file (default: protein fasta)" type="data" format="fasta" optional="false" multiple="false"/>
+    <param name="input_format"
+        label="Genome input format"
+        type="select"
+        optional="true"
+        multiple="false">
+        <option value="" selected="true">Protein FASTA file (default)</option>
+        <option value="--dna">DNA FASTA file</option>
+    </param>
+    <param name="output_format"
+       label="Output format: SBML (default), SBML-fbc2, or old cobra format"
+       type="select"
+       optional="true"
+       multiple="false">
+        <option value="" selected="true">SBML (default)</option>
+        <option value="--fc2">SBML-fbc2</option>
+        <option value="--cobra">Cobra (legacy)</option>
+    </param>
+    <param name="universe"
+        label="Pre-built universe model (default: Bacteria)"
+        type="select"
+        optional="true"
+        multiple="false">
+        <option value="" selected="true">Bacteria (default)</option>
+        <option value="--universe grampos">Gram Positive</option>
+        <option value="--universe gramneg">Gram Negative</option>
+        <option value="--universe cyanobacteria">Cyanobacteria</option>
+        <option value="--universe archaea">Archaea</option>
+    </param>
+    <param name="diamond_args" label="Additional arguments for running diamond"  type="text" optional="true"/>
```

Comment:
That is a bit dangerous, because bad people could try to sneak in other command line arguments. Which params are used a lot?